doc/KUKSA.val_data_broker/authorization.md

```diff
@@ -0,0 +1,505 @@
+| aud       | REQUIRED - as defined in Section 4.1.3 of [RFC7519]. See Section 3 for indications on how an authorization server should determine the value of "aud" depending on the request. |
+| sub       | REQUIRED - as defined in Section 4.1.2 of [RFC7519]. In cases of access tokens obtained through grants where a resource owner is involved, such as the authorization code grant, the value of "sub" SHOULD correspond to the subject identifier of the resource owner. In cases of access tokens obtained through grants where no resource owner is involved, such as the client credentials grant, the value of "sub" SHOULD correspond to an identifier the authorization server uses to indicate the client application. See Section 5 for more details on this scenario. Also, see Section 6 for a discussion about how different choices in assigning "sub" values can impact privacy. |
+| client_id | REQUIRED - as defined in Section 4.3 of [RFC8693].   |
+| iat       | REQUIRED - as defined in Section 4.1.6 of [RFC7519]. This claim identifies the time at which the JWT access token was issued. |
+| jti       | REQUIRED - as defined in Section 4.1.7 of [RFC7519]. |
+
+### Authorization claims
+> If an authorization request includes a scope parameter, the corresponding issued
+> JWT access token SHOULD include a "scope" claim as defined in Section 4.2 of [RFC8693].
+
+> All the individual scope strings in the "scope" claim MUST have meaning for the
+> resources indicated in the "aud" claim. See Section 5 for more considerations about
+>  the relationship between scope strings and resources indicated by the "aud" claim.
+
+| Claim     | Description                                          |
+|-----------|------------------------------------------------------|
+| scope     | The value of the scope claim is a JSON string containing a space-separated list of scopes associated with the token, in the format described in Section 3.3 of [RFC6749] |
+
+## Examples of other API scope implementations
+* https://auth0.com/docs/get-started/apis/scopes/api-scopes
+* https://api.slack.com/scopes
+* https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps
+
+## Implementation in KUKSA.VAL
+A common (perhaps the most common) scenario when delegating access using OAuth2 is that the
+principal owner of a resource has certain access rights that needs to be verified by the
+resource server. This can be accomplished using the user id, group membership, roles or other entitlements present in the access token as claims.
+
+In addition to this, the resource owner can further limit what's authorized by an access
+token by specifying a scope in the request to the authorization server. The issued access
+token will then only grant this subset of authorization to the client (e.g. a third party app)
+using the token.
+
+There are currently no plans in KUKSA.VAL to separately verify the access rights of the
```

Comment:
> If somebody is in possession of a valid token with no scope claim (or empty? is that the same?), he access everything.

Yes, I think that is the idea behind using scope.

It's "traditionally" the case that the resource owner will verify the access rights of the subject _in addition_ to any scope provided. If that is to be added at any point in the future, it would be better to interpret an empty scope as "allow everything" in the current implementation.

With that said, whether an empty scope can even exist in a valid access token is restricted by the authorization server:

>If the client omits the scope parameter when requesting authorization, the authorization server MUST either process the request using a pre-defined default value or fail the request indicating an invalid scope.  The authorization server SHOULD document its scope requirements and default value (if defined). [[RFC6749](https://www.rfc-editor.org/rfc/rfc6749#section-3.3)]

So, in short, this should be better explained in the document.